gen/struct_test.go

```diff
@@ -1856,3 +1857,111 @@ func TestStructLabel(t *testing.T) {
 		})
 	}
 }
+
+func TestNilStructEquals(t *testing.T) {
+	structTypes := []interface{}{
+		te.StructWithOptionalEnum{},
+		tx.DoesNotExistException{},
+		tx.EmptyException{},
+		tu.ArbitraryValue{},
+		tu.Document{},
+		tu.EmptyUnion{},
+		td.DefaultPrimitiveTypedef{},
+		td.Event{},
+		td.Transition{},
+		td.I128{},
+		tf.Cache_Clear_Args{},
+		ts.ContactInfo{},
+		ts.DefaultsStruct{},
+		ts.Edge{},
+		ts.EmptyStruct{},
+		ts.Frame{},
+		ts.GoTags{},
+		ts.Graph{},
+		ts.Node{},
+		ts.Omit{},
+		ts.Point{},
+		ts.PrimitiveOptionalStruct{},
+		ts.PrimitiveRequiredStruct{},
+		ts.Rename{},
+		ts.Size{},
+		ts.StructLabels{},
+		ts.User{},
+		ts.ZapOptOutStruct{},
+		tc.ContainersOfContainers{},
+		tc.EnumContainers{},
+		tc.ListOfConflictingEnums{},
+		tc.ListOfConflictingUUIDs{},
+		tc.MapOfBinaryAndString{},
+		tc.PrimitiveContainers{},
+		tc.PrimitiveContainersRequired{},
+		tf.ConflictingNames_SetValue_Args{},
+		tf.ConflictingNames_SetValue_Result{},
+		tf.KeyValue_GetManyValues_Args{},
+		tf.KeyValue_GetManyValues_Result{},
+		tf.ConflictingNamesSetValueArgs{},
+		tf.InternalError{},
+		tf.KeyValue_SetValueV2_Args{},
+		tf.KeyValue_SetValueV2_Result{},
+		tf.KeyValue_SetValue_Args{},
+		tf.KeyValue_SetValue_Result{},
+		tf.KeyValue_DeleteValue_Args{},
+		tf.KeyValue_DeleteValue_Result{},
+		tf.KeyValue_GetValue_Args{},
+		tf.KeyValue_GetValue_Result{},
+		tf.NonStandardServiceName_NonStandardFunctionName_Args{},
+		tf.NonStandardServiceName_NonStandardFunctionName_Result{},
+		tf.Cache_ClearAfter_Args{},
+		tf.KeyValue_Size_Args{},
+		tf.KeyValue_Size_Result{},
+	}
```

Comment:
Small nit: would `structs` or `instances` be a better name for this slice? We're doing this below:

```
for _, st := range structTypes {
  structType := reflect.TypeOf(st)
```

so I'd expect `structTypes` to be a slice of `reflect.Type`s.

Reply:
In #383, I decided to refer to these as "samples" as-in "a sample of the kind of value I'm testing". Switching to that here.